adapters/33across/33across.go

```diff
@@ -38,39 +55,45 @@ func (a *TtxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters
 	var errs []error
 	var adapterRequests []*adapters.RequestData
 
-	adapterReq, errors := a.makeRequest(request)
-	if adapterReq != nil {
-		adapterRequests = append(adapterRequests, adapterReq)
+	// Construct request extension common to all imps
+	// NOTE: not blocking e
+	TTXRequestExt, err := makeReqExt(request)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	// Break up multi-imp request into multiple external requests since we don't
+	// support SRA in our exchange server
+	for i := 0; i < len(request.Imp); i++ {
+		// makeRequest will either return an adapterRequest or err
+		if adapterReq, err := a.makeRequest(request, request.Imp[i], TTXRequestExt); err == nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		} else {
+			errs = append(errs, err)
+		}
 	}
-	errs = append(errs, errors...)
 
-	return adapterRequests, errors
+	return adapterRequests, errs
 }
 
-// Update the request object to include custom value
-// site.id
-func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
-	var errs []error
+func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest, imp openrtb.Imp, TTXRequestExt []byte) (*adapters.RequestData, error) {
+	var imps []openrtb.Imp
 
-	// Make a copy as we don't want to change the original request
 	reqCopy := *request
-	if err := preprocess(&reqCopy); err != nil {
-		errs = append(errs, err)
+	impCopy, err := makeImps(imp)
+
+	if err != nil {
+		return nil, err
 	}
 
-	if reqCopy.Imp[0].Banner == nil && reqCopy.Imp[0].Video == nil {
-		errs = append(errs, &errortypes.BadInput{
-			Message: "At least one of [banner, video] formats must be defined in Imp. None found",
-		})
+	reqCopy.Imp = append(imps, impCopy)
 
-		return nil, errs
-	}
+	reqCopy.Ext = TTXRequestExt
```

Comment:
Since the `ttxRequestExt` doesn't really change based on imps, I'd suggest moving this outside of this function on L64. You can safely set this as the requestExt for the request sent to your adapter. This request is already a (shallow) copy that is made specifically for your adapter so you can safely modify it unless you're modifying one of the reference/pointer fields